base/deprecated.jl

```diff
@@ -778,6 +778,18 @@ end
 @deprecate cholfact!(A::Base.LinAlg.HermOrSym, uplo::Symbol, ::Type{Val{false}}) cholfact!(A, Val{false})
 @deprecate cholfact!(A::Base.LinAlg.HermOrSym, uplo::Symbol = :U) cholfact!(A)
 
+# #18218
+function arithtype(T)
+    depwarn("arithtype is deprecated as it is no more needed un Julia Base. It will not be available in future release.",
+        :arithtype)
+    T
+end
+function arithtype(::Type{Bool})
+    depwarn("arithtype is deprecated as it is no more needed un Julia Base. It will not be available in future release.",
+        :arithtype)
+    Int
+end
```

Comment:
Those need
```Julia
eval(Base.LinAlg, quote
# ...
end)
```
around them to define `Base.LinAlg.arithtype` as before insead of `Base.arithtype`. Also "is no longer need in".